CONTRIBUTING.md

```diff
@@ -1,3 +1,50 @@
+# Building and Running Tests Locally
+
+To ensure that the Trace Agent works cross-platform, we use TypeScript build scripts located in the [`scripts`](scripts) directory. The entry point to these scripts is [`index.ts`](scripts/index.ts). The usage of this file is `ts-node -p ./scripts ./scripts/index.ts [command1] [...moreCommands]` (assuming that you are running in the repository root directory.)
+
+The list of possible build commands is enumerated as `case` statements in `index.ts`, in addition to `npm-*` commands. See [`index.ts`](scripts/index.ts) for more details.
+
+`npm run script` is an alias for `ts-node -p ./scripts ./scripts/index.ts`.
+
+For example, to compile all scripts and then initialize test fixtures, the command to use would be one of:
+
+```bash
+# Option 1
+ts-node -p ./scripts ./scripts/index.ts npm-compile-all init-test-fixtures
+
+# Option 2
+npm run script npm-compile-all init-test-fixtures
```

Comment:
Don't you need `--` between `script` and the rest of the arguments?

Reply:
We don't need it. I believe `npm run x` is mapped to its underlying script first, and then the rest of the arguments are passed to the result of that evaluation.

Reply:
OK. Maybe I was misunderstanding something.